lib/osf-components/addon/components/addons-service/manager/component.ts

```diff
@@ -33,11 +42,40 @@ export default class AddonsServiceManagerComponent extends Component<Args> {
     node = this.args.node;
     @tracked addonServiceNode?: InternalResourceModel;
 
-    @tracked addonProviders: Provider[] = [];
+    possibleFilterTypes = Object.values(FilterTypes);
+    @tracked filterText = '';
+    @tracked activeFilterType: FilterTypes = FilterTypes.STORAGE;
+    @tracked storageProviders: Provider[] = [];
+    @tracked allAddonProviders: Provider[] = [];
 
     @tracked pageMode?: PageMode;
     @tracked selectedProvider?: Provider;
 
+    @action
+    filterByAddonType(type: FilterTypes) {
+        this.activeFilterType = type;
+    }
+
+    get filteredAddonProviders() {
+        let possibleProviders: any[] = [];
+        switch (this.activeFilterType) {
+        case FilterTypes.STORAGE:
+            possibleProviders = this.storageProviders;
+            break;
+        case FilterTypes.CITATION_MANAGER:
+        case FilterTypes.CLOUD_COMPUTING:
```

Comment:
Should we go ahead and add our three addons that fit into these categories and handle these cases? Or do we wait because there's no matching backend yet?

Reply:
I think we could probably add the `ExternalStorageProvider` equivalent for these addon types. This would actually help me test out some of the loading states a little better too.